tensorflow/core/lib/strings/str_util.h

```diff
@@ -131,6 +131,15 @@ std::vector<string> Split(StringPiece text, StringPiece delims);
 template <typename Predicate>
 std::vector<string> Split(StringPiece text, StringPiece delims, Predicate p);
 
+// Split "text" into into a vector of strings where each string
+// in the array is a string of ASII or unicode chars. If the
+// delim is empty, then each string in the array consists of
+// one ASCII or one unicode char. If successful, the result
+// is passed to "*result" and returns true. Otherwise returns false
+// and error message is passed to "*error".
+bool SplitUTF8(StringPiece text, string delim, std::vector<string>* result,
```

Comment:
Can we use the `Status` object to return errors?

Also, you should use `const string& delim`, or `StringPiece delim`. Isn't there a restriction that `delim` is only one character? If so, we should document that, and check for that in the implementation.

Reply:
Thanks @drpngx. The PR has been updated with `Status` returned and `const string& delim`.

For `delim` length, in the kernel of StringSplitOp, there is already a check to guard the length of `delim`.

In the util library (`str_util.h`) the `SplitUTF8` actually allows a UTF8 encoded string (1-4 chars) as the delim. There is a test case in `str_util_test.cc` to show that.